content/en/docs/configuration/acme/dns01/azuredns.md

```diff
@@ -5,6 +5,113 @@ weight: 30
 type: "docs"
 ---
 
+To configure the AzureDNS DNS01 Challenge in a Kubernetes cluster there are 2 ways:
+- A Managed Identity
+- Service Principal
+
+## Managed Identity
+
+- First of all we need to create the identity in Azure Cloud (the below code is Terraform, but this action can be performed also with `azure-cli`)
+
+> The resource group must be already existing in order for this to work.
+> This role allows `cert-manager` to control `AzureDNS` zones in the same resource group as the identity lives in
+
+```terraform
+variable "cluster_name" {
+  type = string
+}
+variable resource_group_name {}
+variable resource_group_id {}
+variable location {}
+
+output client_id {
+  value = azurerm_user_assigned_identity.main.client_id
+}
+
+resource "azurerm_user_assigned_identity" "main" {
+  name                = "${var.cluster_name}-cert-manager"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+}
+
+resource "azurerm_role_definition" "main" {
+  name        = "${var.cluster_name}-cert-manager"
+  scope       = var.resource_group_id
+  description = "This role is used by cert-manager"
+
+  permissions {
+    actions = [
+      "Microsoft.Network/dnsZones/*/read",
+      "Microsoft.Network/dnsZones/*/write",
+      "Microsoft.Network/dnsZones/*/delete",
+      "Microsoft.Network/dnsZones/read",
+    ]
+  }
+  assignable_scopes = [
+    var.resource_group_id
+  ]
+}
+
+resource "azurerm_role_assignment" "main" {
```

Comment:
Can this role assignment be just on the DNS Zone and can it use the pre-canned `DNS Zone contributor` role?

```
resource "azurerm_role_assignment" "main" {
  role_name    = "DNS Zone contributor"
  principal_id = azurerm_user_assigned_identity.main.principal_id
  scope        = azurerm_dns_zone.example.id
}
```